CONTRIBUTING.md

```diff
@@ -32,6 +32,9 @@ To check that all codes are tested and documented there's a test that error code
 ## Test generator
 Tests are automatically generated for files in the `tests/eval_files/` directory, with the code that it's testing interpreted from the file name. The file extension is split off, if there's a match for for `_py\d*` it strips that off and uses it to determine if there's a minimum python version for which the test should only run.
 
+### autofix files
+Checks that have autofixing can have a file in the `tests/autofix_files` directory matching the filename in `tests/eval_files`. The result of running the checker on the eval file with autofix enabled will then be compared to the content of the autofix file and will print a diff (if `-s` is on) and assert that the content is the same. `--generate-autofix` is added as a pytest flag to ease development, which will print a diff (with `-s`) and overwrite the content of the autofix file. Also see the magic line marker `pass # AUTOFIX_LINE ` below
```

Comment:
Instead of spitting out the resulting `.py` files, can we save the diff between before and after as a `.py.diff`?  It's much easier to get a sense of "what it's doing" in this format, and I'm concerned that without that code review is not going to be effective.

(although instead of constructing a three-way diff, I'd just let Pytest's assertion helper show the diff between the actual and expected diffs)

((diff diff diff, it hardly seems to ~~diff~~ mean anything anymore.  diff)

Reply:
I like the generated files for checking that they're valid and passes linters, and when developing they're easier to read and throw into `cst.parse_module`, but I'll generate diff files as well :+1:

Reply:
Oh, yep, both sounds good!

Reply:
Wohoo, bigger commits! :grin:
They're still not perfect, as they're not showing e.g. where checkpoints *aren't* inserted - but that will be covered by future tests that autofixed files don't generate any errors. (and will have to figure out a solution for errors that are expected not to get autofixed)